be/test/exec/vectorized/json_scanner_test.cpp

```diff
@@ -601,4 +601,47 @@ TEST_F(JsonScannerTest, test_native_json_single_column) {
               chunk->debug_row(4));
 }
 
+// simdjson::ondemand::object cannot be iterated after field accessing.
+// See https://github.com/StarRocks/starrocks/issues/3648.
+TEST_F(JsonScannerTest, test_native_json_single_column_2) {
+    std::vector<TypeDescriptor> types;
+    constexpr int kColumns = 2;
+    for (int i = 0; i < kColumns; i++) {
+        types.emplace_back(TypeDescriptor::create_json_type());
+    }
+
+    std::vector<TBrokerRangeDesc> ranges;
+    TBrokerRangeDesc range;
+    range.format_type = TFileFormatType::FORMAT_JSON;
+    range.strip_outer_array = false;
+    range.__isset.strip_outer_array = false;
+    range.__isset.jsonpaths = true;
+    // Iteration after field accessing.
+    range.jsonpaths = R"(["$.k1", "$"])";
```

Comment:
well, how about make it as parameters, and extract the test case to a common function?